Add April newsletter (#188)
Financial-Times/origami-website: _posts/2020-05-01-newsletter.md

---
title: Origami Newsletter, April 2020
description: This issue features the Origami Component Survey, Automating Visual Testing of Origami Components and moving from CircleCI to GitHub Actions
author: Jake Champion
tags:
  - Newsletter
---

<abbr title="Too long; didn't read">
	<strong>
	TL;DR:
	</strong>
</abbr> {{page.description}}

## Top Things

These are some of the bigger things we've done over the last month.

### Origami Components Survey

> This quarter we are aiming to find out what our users know about Origami, how much of Origami they use and how we can improve our products for all of our users.

Reviewer: I think maybe you're thinking of the upcoming specialist survey here? The survey that went out was much more focused on understanding the barriers that teams have to developing their own Origami (or Origami-compatible) components

Reviewer: ah yeah, sorry somehow I misinterpreted it

Our first survey has already gone out, it was to find out how we can help teams build their own Origami Components. You can read more about the survey results [on our blog](https://origami.ft.com/blog/2020/04/20/origami-survey-results/).

### Automated Visual Regression Testing

When reviewing changes made to an Origami Component, we have to view the component in multiple browsers, at mulitple resolutions and mulitple different configurations for the component. This is to check if any visual bugs have been introduced accidentally as part of the change. It is a rather taxing and manual piece of work.

> We've automated all of that manual work now. Now when a change is made to an Origami Component, if the change has caused any of the visual aspects of the component to be different we are made aware of those changes and can quickly review and approve the work.

Reviewer: 🙌

This work has already proved to be useful by immediately finding a visual regression for a change in o-teaser, which we were able to fix before it went live on any website.

<img width="980" alt="Screenshot which shows the visual regression tooling working on the o-teaser component" src="https://user-images.githubusercontent.com/1569131/80602987-bc8df900-8a27-11ea-84b2-1329c45a3d3b.png">

### Moved from CircleCI to GitHub Actions

> We've moved all of our automated tooling which was on CircleCI to GitHub Actions. This should free shorten the size of the CircleCI queue for every FT project, making their CI feedback loop shorter.

Reviewer: ```suggestion
We've moved all of our automated tooling which was on CircleCI to GitHub Actions. This should shorten the size of the CircleCI queue for every FT project, making our CI feedback loop shorter.
```

> Moving to GitHub Actions has helped us manage all our repositories by making it simple to automate types of work such as releasing changes, adding tickets to our project board.

Reviewer: ```suggestion
Moving to GitHub Actions has helped us manage all our repositories by making it simple to automate types of work such as releasing changes and adding tickets to our project board.
```

As part of this work we've also made a [GitHub Action for the Operations & Reliability Change API](https://github.com/Financial-Times/change-api-action/).

## Special Thanks

> This months special thanks goes to TODO

Author: ```suggestion
This months special thanks goes to all the people who filled in our first survey. You all are helping to improve Origami, thank you.
```

## Broader Update

A digest list of other things that have happened since our last update:

> TODO

Author: @notlee How do you fill out this section with so much stuff?!

Reviewer: manually 😬 I skim the project board for repos that have had changes, then do a quick summary of any releases over the last month with a shoutout if someone other than us contributed

Reviewer: we should look at automating this in some way now we're doing regular newsletters. I don't think sending release logs verbatim would be very helpful, but at least to give us a list to write a summary for

Reviewer 2: we do love labels. what if we attach a label of `newsletter` to releases that we feel, at the time, are newsworthy

Reviewer 2: then when it gets to time to write the newsletter you can do label:newsletter

Reviewer 2: we can add it when we're adding the `release:` label ^^

Author: Good idea @chee 

These are all the ones I've found for this month --
https://github.com/Financial-Times/o-share/pull/165
https://github.com/Financial-Times/o-share/pull/168
https://github.com/Financial-Times/o-share/pull/169
https://github.com/Financial-Times/o-share/pull/170
https://github.com/Financial-Times/polyfill-library/issues/536
https://github.com/Financial-Times/polyfill-library/issues/50
https://github.com/Financial-Times/polyfill-library/pull/559
https://github.com/Financial-Times/polyfill-library/pull/561
https://github.com/Financial-Times/o-layout/pull/143
https://github.com/Financial-Times/polyfill-library/pull/562
https://github.com/Financial-Times/o-forms/pull/336
https://github.com/Financial-Times/o-forms/pull/336
https://github.com/Financial-Times/o-forms/pull/337
https://github.com/Financial-Times/o-quote/pull/55
https://github.com/Financial-Times/o-colors/pull/255
https://github.com/Financial-Times/o-colors/pull/257
https://github.com/Financial-Times/o-expander/pull/93
https://github.com/Financial-Times/polyfill-library/pull/572
https://github.com/Financial-Times/o-buttons/pull/248
https://github.com/Financial-Times/o-buttons/pull/251
https://github.com/Financial-Times/o-grid/pull/211
https://github.com/Financial-Times/polyfill-library/pull/569
https://github.com/Financial-Times/polyfill-library/pull/570
https://github.com/Financial-Times/o-teaser/pull/162
https://github.com/Financial-Times/o-topper/pull/44
https://github.com/Financial-Times/o-typography/pull/262
https://github.com/Financial-Times/o-editorial-layout/pull/14
https://github.com/Financial-Times/o-forms/pull/341
https://github.com/Financial-Times/o-syntax-highlight/pull/51
https://github.com/Financial-Times/o-typography/pull/263
https://github.com/Financial-Times/o-editorial-typography/pull/28
https://github.com/Financial-Times/o-teaser/pull/164
https://github.com/Financial-Times/o-table/pull/197

o-share also got improved accessbility and analytics support thanks to Glynn Phillips in Customer Products. -- https://github.com/Financial-Times/o-share/pull/168 https://github.com/Financial-Times/o-share/pull/170

polyfill-library fixed a bug for Map support in Internet Explorer -- https://github.com/Financial-Times/polyfill-library/issues/536

polyfill-library also added Object.fromEntries support in Internet Explorer -- https://github.com/Financial-Times/polyfill-library/issues/536

polyfill-library fixed a bug in HTMLCanvasElement.prototype.toBlob support for Edge -- https://github.com/Financial-Times/polyfill-library/pull/561

polyfill-library added URLSearchParams.sort method support for all browsers -- https://github.com/Financial-Times/polyfill-library/pull/562

polyfill-librarb added CSS.Supports support for all browsers -- https://github.com/Financial-Times/polyfill-library/pull/569

> o-quote now has an editorial style <img src='https://user-images.githubusercontent.com/10405691/78697295-82ce4480-78f8-11ea-99af-278ea5eae152.png' alt='the new o-quote editorial design' /> -- https://github.com/Financial-Times/o-quote/pull/55

Reviewer: shout out to Josh from the Content Innovation team for the detail work

> o-quote now has an editorial style <img src='https://user-images.githubusercontent.com/10405691/78697295-82ce4480-78f8-11ea-99af-278ea5eae152.png' alt='the new o-quote editorial design' /> -- https://github.com/Financial-Times/o-quote/pull/55

Author: ```suggestion
o-quote now has an editorial style thanks to Josh from the Content Innovation team.
<img src='https://user-images.githubusercontent.com/10405691/78697295-82ce4480-78f8-11ea-99af-278ea5eae152.png' alt='the new o-quote editorial design' />
```

<img src='https://user-images.githubusercontent.com/10405691/78697295-82ce4480-78f8-11ea-99af-278ea5eae152.png' alt='the new o-quote editorial design' />

o-editorial-layout uses the new editorial quote style from o-quote <img src='https://user-images.githubusercontent.com/10405691/80201001-5ebe7300-861b-11ea-98c1-8c4448db4f25.png' alt='new editorial quote style in o-editorial-layout' /> -- https://github.com/Financial-Times/o-editorial-layout/pull/14

> o-teaser uses a new design which updated the spacing -- https://github.com/Financial-Times/o-teaser/pull/164

Author: ```suggestion
o-teaser uses a new design which updated the spacing thanks to Eray and Mark from the Apps team.
```

> o-teaser uses a new design which updated the spacing -- https://github.com/Financial-Times/o-teaser/pull/164

Reviewer: shout out to Eray and Mark from the apps team for their teaser work
